server/planning/commands/purge_expired_locks.py

```diff
@@ -39,109 +65,103 @@ class PurgeExpiredLocks(Command):
         $ python manage.py planning:purge_expired_locks -r all
         $ python manage.py planning:purge_expired_locks -r all -e 48
     """
+    return await purge_expired_locks_handler(resource, expire_hours)
 
-    option_list = [
-        Option("--resource", "-r", required=True),
-        Option("--expire-hours", "-e", dest="expire_hours", required=False, type=int, default=24),
-    ]
 
-    def run(self, resource: str, expire_hours: int = 24) -> None:
-        logger.info("Starting to purge expired item locks")
+async def purge_expired_locks_handler(resource: str, expire_hours: int = 24):
+    logger.info("Starting to purge expired item locks")
 
-        if resource == "all":
-            resources = ["events", "planning", "assignments"]
-        elif resource not in ["events", "planning", "assignments"]:
-            raise ValueError(f"Invalid resource: {resource}")
-        else:
-            resources = [resource]
+    if resource == "all":
+        resources = ["events", "planning", "assignments"]
+    elif resource not in ["events", "planning", "assignments"]:
+        raise ValueError(f"Invalid resource: {resource}")
+    else:
+        resources = [resource]
+
+    lock_name = get_lock_id("purge_expired_locks", resource)
+    if not lock(lock_name, expire=600):
+        logger.info("purge expired locks task is already running")
+        return
+
+    expiry_datetime = date_to_str(utcnow() - timedelta(hours=expire_hours))
+    for resource_name in resources:
+        try:
+            await purge_item_locks(resource_name, expiry_datetime)
+        except Exception as err:
+            logger.exception(f"Failed to purge item locks ({err})")
+
+    unlock(lock_name)
+    logger.info("Completed purging expired item locks")
+
+
+async def purge_item_locks(resource: str, expiry_datetime: str):
+    logger.info(f"Purging expired locks for {resource}")
+    resource_service = SERVICE_MAPPING[resource]
+    try:
+        autosave_service = get_resource_service("event_autosave" if resource == "events" else f"{resource}_autosave")
```

Comment:
@MarkLark86 we are making use of autosave services here. They are yet to be converted to async. Do they remain as is and just add a TODO to get back to them when finished ?